app.rb

```diff
@@ -6,20 +6,40 @@
 require_relative './lib/players/human.rb'
 require_relative './lib/ui/user_interface.rb'
 require_relative './lib/ui/io.rb'
+require_relative './lib/scripts/setup_script.rb'
+require_relative './lib/validators/setup_validator.rb'
 
-def initializer(size, token1, token2)
+def initializer(size)
+  player1 = :'Player 1'
+  player2 = :'Player 2'
   board = Board.new(size, '-')
   checker = BoardChecker.new(3, '-')
   io = IO.new(STDIN, STDOUT)
-  user_interface = UserInterface.new(io, board, token1, token2)
+  user_interface = UserInterface.new(io, board)
+  setup_script = SetupScript.new
+  setup_validator = SetupValidator.new(setup_script)
+
+  user_interface.welcome
+  token1 = select_token(user_interface, setup_script, setup_validator, player1, '')
+  token2 = select_token(user_interface, setup_script, setup_validator, player2, token1)
```

Comment:
I think it's probably worth breaking out a GameSetup class and a Configuration object at this point, instead of some helper functions. That way inside of this game intializer we don't have any real logic, all it needs to do is wire everything up, welcoming the user and setting up the tokens can happen elsewhere.

Reply:
Now have a game setup class with a configuration struct.